tests/backup/backup_helper.go

```diff
@@ -1500,7 +1505,87 @@ func IsPresent(dataSlice interface{}, data interface{}) bool {
 		}
 	}
 	return false
+}
+
+// getStorkImageVersion returns current stork image version.
+func getStorkImageVersion() (string, error) {
+	storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return "", err
+	}
+	storkImage := storkDeployment.Spec.Template.Spec.Containers[0].Image
+	storkImageVersion := strings.Split(storkImage, ":")[len(strings.Split(storkImage, ":"))-1]
+	return storkImageVersion, nil
+}
+
+// upgradeStorkVersion upgrades the stork to the provided version.
+func upgradeStorkVersion(storkImageToUpgrade string) (bool, error) {
+	var upgradeStatus bool
+	currentStorkImageStr, err := getStorkImageVersion()
+	if err != nil {
+		return false, err
+	}
+	currentStorkVersion, _ := version.NewSemver(currentStorkImageStr)
+
+	storkImageVersionToUpgradeStr := strings.Split(storkImageToUpgrade, ":")[len(strings.Split(storkImageToUpgrade, ":"))-1]
+	storkImageVersionToUpgrade, _ := version.NewSemver(storkImageVersionToUpgradeStr)
+
+	log.Infof("Current stork version : %s", currentStorkVersion)
+	log.Infof("Upgrading stork version to : %s", storkImageVersionToUpgrade)
+
+	if currentStorkVersion.GreaterThanOrEqual(storkImageVersionToUpgrade) {
+		log.Errorf("Cannot upgrade stork version from %s to %s as the current version is higher than the provided version", currentStorkVersion, storkImageVersionToUpgrade)
+		return false, nil
+	}
+
+	isOpBased, _ := Inst().V.IsOperatorBasedInstall()
+	if isOpBased {
+		storageSpec, err := Inst().V.GetDriver()
+		if err != nil {
+			return false, err
+		}
+		storageSpec.Spec.Stork.Image = storkImageToUpgrade
+		_, err = operator.Instance().UpdateStorageCluster(storageSpec)
+		if err != nil {
+			return false, err
+		}
+	} else {
+		storkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+		if err != nil {
+			return false, err
+		}
 
+		storkDeployment.Spec.Template.Spec.Containers[0].Image = storkImageToUpgrade
+		_, err = apps.Instance().UpdateDeployment(storkDeployment)
+		if err != nil {
+			return false, err
+		}
+	}
+
+	// validate stork pods after upgrade
+	time.Sleep((1 * time.Minute))
+	updatedStorkDeployment, err := apps.Instance().GetDeployment(storkDeploymentName, storkDeploymentNamespace)
+	if err != nil {
+		return false, err
+	}
+	err = apps.Instance().ValidateDeployment(updatedStorkDeployment, k8s.DefaultTimeout, k8s.DefaultRetryInterval)
+	if err != nil {
+		return false, err
+	}
+	deployStatus := updatedStorkDeployment.Status.Conditions[1].Status
+	latestStorkImageVersionStr, _ := getStorkImageVersion()
+	if err != nil {
+		return false, err
+	}
+	latestStorkImageVersion, _ := version.NewSemver(latestStorkImageVersionStr)
```

Comment:
If the version.NewSemver function returns an error, how about we handle it?

Reply:
Done